Doc installation guide (#116)
nest-desktop/nest-desktop: docs/user/setup.rst

===========

This guide provides a detailed documentation on how to install NEST Desktop with NEST Simulator.
The front end NEST Desktop requires NEST Simulator as back end for the simulation engine.

.. image:: ../_static/img/installation-guide.png
  :width: 100%
  :target: #setup-guide

|

.. note::

>   NEST Simulator has API Server which can forward requests to simulation engine.
>   To read more information about API Server of the NEST Simulator, `here <https://nest-simulator.readthedocs.io/en/stable/nest_server.html>`__.

Reviewer: ```suggestion
  NEST Simulator has an API Server which can forward requests to simulation engine.
  For more information about the API Server of the NEST Simulator, please have a look`here <https://nest-simulator.readthedocs.io/en/stable/nest_server.html>`__.
```

.. code-block:: bash

  apt install docker.io docker-compose

2. Get configuration file for Docker-compose (`docker-compose.yml <https://raw.githubusercontent.com/nest-desktop/nest-desktop/main/docker-compose.yml>`__)

.. code-block:: bash

  wget https://raw.githubusercontent.com/nest-desktop/nest-desktop/main/docker-compose.yml

3. Build and start NEST Desktop and NEST Simulator with a single command:

.. code-block:: bash

  docker-compose up --build

NEST Desktop is now serving at ``http://localhost:8000`` and NEST Simulator at ``http://localhost:5000``.

For more information read the full documentation of the `NEST Desktop Docker <https://github.com/nest-desktop/nest-desktop-docker>`__.

Via Singularity
---------------

.. image:: ../_static/img/logo/singularity-logo.svg
  :width: 240px
  :target: #via-singularity

|

Singularity is an application container for **Linux** systems.
For more information read the full documentation of the Singularity
`here <https://sylabs.io/docs/>`__.

1. Clone a working copy from the repository and go to the folder:

.. code-block:: bash

  git clone https://github.com/nest-desktop/nest-desktop-singularity
  cd nest-desktop-singularity

> 2. Register bash command for NEST Desktop Singularity:

Reviewer: ```suggestion
2. Register the bash command for NEST Desktop Singularity:
```

.. code-block:: bash

  export PATH=$PATH:$PWD/bin/

> 3. Build Singularity images (it will ask for sudo password):

Reviewer: ```suggestion
3. Build the Singularity images (it will ask for sudo password):
```

.. code-block:: bash

  nest-desktop-singularity build

> 4. Start Singularity instances of NEST Desktop and NEST Simulator:

Reviewer: ```suggestion
4. Start the Singularity instances of NEST Desktop and NEST Simulator:
```

.. code-block:: bash

  nest-desktop-singularity start

NEST Desktop is now serving at ``http://localhost:8000`` and NEST Simulator at ``http://localhost:5000``.

For more information read the full documentation of the `NEST Desktop Singularity <https://github.com/nest-desktop/nest-desktop-singularity>`__.

Via Python Package
------------------

.. image:: ../_static/img/logo/pypi-logo.svg
  :width: 240px
  :target: #via-python-package

|

.. note::

>   This section shows setup guide for only NEST Desktop.

Reviewer: ```suggestion
  This section shows the setup guide for NEST Desktop only.
```

  NEST Simulator cannot be installed via pip (`maybe soon <https://github.com/nest/nest-simulator/pull/2073>`__).
  For more information read the full installation guide of NEST Simulator
  `here <https://nest-simulator.readthedocs.io/en/latest/installation/index.html>`__.

1. NEST Desktop is available on PyPI and can be installed with ``pip`` command:

.. code-block:: bash

  pip3 install nest-desktop [--user] [--upgrade]

> For more information read the full installing docs :doc:`here <setup>`.

Reviewer: ```suggestion
For more information read the complete installation docs :doc:`here <setup>`.
```
